src/kdtree.rs

```diff
@@ -209,13 +207,12 @@ impl<'a, T> KdTree<'a, T> {
     }
 
     fn extend(&mut self, point: &[f64]) {
-        for dim in 0..self.dimensions {
-            if self.min_bounds[dim].is_nan() || self.min_bounds[dim] > point[dim] {
-                self.min_bounds[dim] = point[dim];
-            }
-            if self.max_bounds[dim].is_nan() || self.max_bounds[dim] < point[dim] {
-                self.max_bounds[dim] = point[dim];
-            }
+        let (mut i, mut a, mut d) = 
```

Comment:
This will truncate at the minimum of iterator lengths (which *should* be equal to `self.dimensions`).